REPL.cpp

```diff
@@ -2,18 +2,65 @@
 #include "Logging.h"
 #include "TransformAST.h"
 
+#include <cstdint>
+#include <tuple>
+#include <type_traits>
+
+#include <swift/AST/ASTMangler.h>
+
 #define SWIFT_MODULE_PATH "S:\\b\\swift\\lib\\swift\\windows\\x86_64"
 
-void ConfigureFunctionLinkage(std::unique_ptr<swift::SILModule> &sil_module)
+// Returns expression function and result global variable
+std::tuple<swift::FuncDecl *, swift::VarDecl *> ConfigureFunctionLinkage(swift::SourceFile &src_file, std::unique_ptr<swift::SILModule> &sil_module)
 {
     SetCurrentLoggingArea(LoggingArea::SIL);
-    for(auto &fn : sil_module->getFunctionList())
+
+    swift::FuncDecl *res_fn = nullptr;
+    swift::VarDecl *res_var = nullptr;
+
+    const std::string res_fn_name = src_file.getFilename().str();
+    for(swift::Decl *decl : src_file.Decls)
     {
-        fn.setLinkage(swift::SILLinkage::Public);
-        Log(std::string("Set SIL Function " + fn.getName().str() + " to public"));
-    }
+        assert(decl);
+        if(auto fn = llvm::dyn_cast<swift::FuncDecl>(decl))
+        {
+            swift::SILDeclRef sil_decl(fn);
+
+            std::string name_original = fn->getName().str();
+            std::string name_mangled = sil_decl.mangle();
 
+            sil_module->lookUpFunction(sil_decl)->setLinkage(swift::SILLinkage::Public);
+            Log(std::string("Set function ") + name_original + " (" + name_mangled + ") to public");
+
+            if(name_original == res_fn_name)
+                res_fn = fn;
+        }
+        else if(auto var = llvm::dyn_cast<swift::VarDecl>(decl))
+        {
+            std::string name = var->getNameStr();
+            if(name == res_fn_name + "_res")
+                res_var = var;
+        }
+    }
+    assert((res_fn == nullptr && res_var == nullptr) ||
+           (res_fn != nullptr && res_var != nullptr));
     sil_module->lookUpFunction("main")->setLinkage(swift::SILLinkage::Private);
+    return std::make_tuple(res_fn, res_var);
+}
+
+llvm::Expected<std::unique_ptr<REPL>> REPL::Create()
+{
+    std::unique_ptr<REPL> result(new REPL);
+    auto jit = JIT::Create();
+    SetCurrentLoggingArea(LoggingArea::JIT);
+    if(!jit)
+    {
+        llvm::Error err(std::move(jit.takeError()));
```

Comment:
You don't need to make a new error. Just propagate it below: `return jit.takeError();`